### PR TITLE
Functional and Non-Functional Requirements

### PR DESCRIPTION
**Pull request by FSXAC**
- Upgraded enumerate tags to use the newer "enumitem" package
- Added functional requirements for RPAS
- Added non-functional requirements for RPAS
- Added citations for resources used for requirements

### CODE REVIEW

File: docs/reqs/reqs.tex [hunk 1]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
-    \item The drone should have systems for low-battery alarm and emergency protocol for low-battery and loss of control radio.
-\end{enumerate}
-
-\subsection{System Reliability and Durability}
-\begin{enumerate}[{NF.SR}.1:]
-    \item Brushless DC Motors used on the drone should have a life span of 10,000 flight-hours.
-    \item Non-computation-platform electronics are expected to have a life span of 20,000 flight hours if the procedures are followed correctly
-    \item The lithium polymer battery for the drone is expected to retain 75\% of its charge after 200 charge/discharge cycles. (TODO: make citation \url{http://learningrc.com/lipo-battery/}) 
+\begin{enumerate}[label=NF.DR.\arabic*, wide=1cm, widest=3cm, leftmargin=*, font=\bfseries, noitemsep,topsep=0pt, parsep=4pt, partopsep=0pt]
+	\item The multirotor RPAS shall be able to carry a maximum payload mass of 2.0 kg or 4.4 lb and achieve flight.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: Stick to metric here, no point in citing both

File: docs/reqs/reqs.tex [hunk 2]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
+	\item The multirotor RPAS maximum take off mass including the payload shall be less than 25.0 kg in compliance with Transport Canada TC-15263.\cite{tp15263}
+	\item The multirotor RPAS shall withstand 5 nautical miles (knots) of crosswind during operation.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: Like above, stick to knots. Also, how do we verify this?

File: docs/reqs/reqs.tex [hunk 3]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
+	\item The multirotor RPAS shall receive pilot TX signals via 2.4 GHz radio from 300 m or 1000 ft away.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: In a different requirement we constrain ourselves to 30 meters for WiFi (different system, I know, but same overall device), so I don't know if that conflicts with this.

File: docs/reqs/reqs.tex [hunk 4]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
+	\item The multirotor RPAS orientation LEDs shall be clearly visible from 300 m or 1000 ft away.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: What orientation LEDs? Specify.

File: docs/reqs/reqs.tex [hunk 5]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
+	\item The multirotor RPAS flight controllers systems shall respond to pilot controls with less than 100ms of latency.
+	\item The multirotor RPAS shall have a hovering flight duration of 30 minutes with 1.0 kg of payload.
+	\item The brushless DC (BLDC) motors shall have a kV constant between 800 and 1200.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: Need to add kV to the glossary

File: docs/reqs/reqs.tex [hunk 6]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
+	\item The BLDC motors and propellers shall output at least 8,000 N of thrust at maximum throttle.
+	\item The BLDC motors and propellers shall have an optimal efficiency of at least 50 N/W.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: I don't think this is necessary

File: docs/reqs/reqs.tex [hunk 7]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
+	\item The BLDC motors and propellers shall have a combined mass of less than 100g.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: Why is this strictly required?

File: docs/reqs/reqs.tex [hunk 8]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
+	\item The electronic speed controller (ESC) for controlling BLDC motor speeds shall have maximum current output of 30 A.
+	\item The ESC shall not exceed operating temperature of 60 degrees Celsius at maximum load operation.
+	\item The ESC shall operate at maximum load --- at 90\% of the maximum current draw --- for 60 seconds.
+	\item The multirotor RPAS battery shall be a high-power-output Lithium Polymer (LiPo) battery with 5000 mAh to 6000 mAh capacity and discharge rating of at least 30 C to 40 C.
+	\item The multirotor RPAS battery capacity shall not exceed 6700 mAh.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: Why 6700mAh?

File: docs/reqs/reqs.tex [hunk 9]
```diff
@@ -192,118 +192,130 @@ \section{Goal}
...
+	\item The LiPo batteries shall retain 75\% of its initial maximum capacity after 200 charge-discharge cycles.\cite{lipobattery}
+	\item The low-battery warning system shall trigger the flight controller to enter an emergency landing protocol when battery is at or below 5\% of the total charge capacity.
+	\item The flight controller shall automatically enter emergency landing protocol when pilot radio control feed has been cut for more than 1.0 seconds with a maximum descend rate of 0.5 m/s.
+	\item The multirotor RPAS airframe mass shall not exceed 400 g.
```
Comment (peterdeutsch) [inline docs/reqs/reqs.tex]: Why is this strictly required?

File: docs/reqs/reqs.tex (hunk 1 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: Ok, I’ll correct it when I get on the computer tomorrow

File: docs/reqs/reqs.tex (hunk 2 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: Knots is a shorthand for nautical miles. I think we need this when mentioning the first time. To verify, fly it when it’s windy outside.

File: docs/reqs/reqs.tex (hunk 3 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: The radio for the flight control is independent of wifi data. Even if we lose the downlink for the wifi, the drone should still be controllable for 3-5 times the range.

File: docs/reqs/reqs.tex (hunk 4 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: Check the functional requirements section on LED orientation indication. I will reword this to make it sound less mystic. But I think the detail on the purpose of this should not be in this document

File: docs/reqs/reqs.tex (hunk 5 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: Ok. kV is a standard design/data sheet parameter that models the linear relationship between motor speed and voltage

File: docs/reqs/reqs.tex (hunk 6 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: It is

File: docs/reqs/reqs.tex (hunk 7 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: Overweight of the motor assembly means less capacity for the payload. technically nothing in the non-functional requirements are strictly required

File: docs/reqs/reqs.tex (hunk 8 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: Because the maximum capacity for a lithium battery allowed on commercial flight is 100Wh, 100Wh/14.8V = 6700mAh. 

Although on second thought perhaps this is better as a constraint

File: docs/reqs/reqs.tex (hunk 9 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: Same as above. Airframe should be as light as possible without sacrificing structural integrity. More weight = less payload = less flight time = less maneuverability. 400g is a good target to hit

File: docs/reqs/reqs.tex (hunk 6 above)
Comment (FSXAC) [reply docs/reqs/reqs.tex]: Let me elaborate, the datasheets I've found where I did my rough calculations from has a motor thrust-to power efficiency in the range of 4.0 to 6.0. 

The faster the motor spins, the faster the propeller is cutting through air, the higher the air resistance (drag is a function of speed^2). So an increase in motor power consumption has a diminishing return on force output.

A motor on full-power can probably output 7000N of thrust (714g if we divide 9.81m/s^2) at a efficiency below 4.0 g/W. Where a motor on half-power could output 3500N (357g) of thrust at a efficiency of about 5.0-6.0 g/W. At very low-power, the motor will only be able to lift a little, but the power is so small the ratio/efficiency is high.

Ideally, the drone should __hover__ (not climb, or with max. take-off payload) where the motors are operating at their optimal efficiency. Which is around 5-6g/W or at least 50 N/W. The reason I chose "hover" is because that is the __typical__ mode of operation.

File: docs/reqs/reqs.tex (hunk 2 above)
Comment (peterdeutsch) [reply docs/reqs/reqs.tex]: Sure, but I don't know if we need to be this specific. We won't be able to test whether our drone withstands 4 knots vs 6 knots.

File: docs/reqs/reqs.tex (hunk 4 above)
Comment (peterdeutsch) [reply docs/reqs/reqs.tex]: I didn't see the functional requirements for this, whoops

File: docs/reqs/reqs.tex (hunk 6 above)
Comment (peterdeutsch) [reply docs/reqs/reqs.tex]: Maybe I should clarify my complaint :)

Yes, we need to lay out these values, but I'm not sure if the requirements document is the place to do it. Mieszko doesn't care about the kV constants of the BLDC motors, and I'd imagine these values are fairly specific to the drone implementation we choose.

Comment (FSXAC) [reply docs/reqs/reqs.tex]: That's a good point. I think for now we'll leave it in. We should really start working on the other documents. If such "internal requirements" make is in the other docs, I guess we can remove it from here.

But for now, I think requirements are still requirements, even if the requirement is invisible to the end user
